### DISCUSSION TITLE
Potential Repo Restructure

### DISCUSSION THREAD

**Discussion by steven-murray** [HERA-Team/hera-validation#72] [Ideas]
We talked briefly about potentially restructuring the repo around different "projects" -- i.e. to have folders for "H1C IDR2", "Simulator Validation", "H3C IDR2" etc. 

I agree in some way this would be more obvious, but at the same time, the current layout is sufficiently general that it can account for these projects. As a recap, the current structure is such that folders within the repo represent conceptual steps of validation (that should be appropriate for any given "project"). Inside these folders are specific kinds of tests, some of which are appropriate for some data releases, and some potentially not. Inside these are the specific runs of tests -- which may or may not need to be iterated for each new data release.

Each _proposed_ test is created as an issue, and flagged with a _Milestone_ that specifies the project(s) that it is a part of, so we can quickly narrow down our tests by project. We also collect all proposed tests in Github Projects according to which "Step" they belong to, which can help filter along those lines. I am less sure of the utility of these projects, since these projects would never be "complete" (even if we finish all tests related to a given step, there's no saying that we wouldn't have to iterate on them for a new data release). 

Despite being general enough, I am open to restructuring. However, one other downside to consider for restructuring is that links in the H1C IDR2 paper to specific tests may become dead if files are moved around. 

In any case, I think it's worthwhile opening the floor for discussion along these lines. Go!

**Comment by piyanatk** [comment]
Are we anticipating the validation steps to differ for different data release? If so, I think organising the repo into folders based on the data release is better, especially as our data grow. But I think the data release folder should be treated as "archival", that is we keep the `test-series` directory as is, keep adding more steps if needed, and do "ongoing" tests in these folders. Then, once the validation for each data release is done, we copy specific commits of the notebooks from the `test-series` to, e.g., `test-H1C_IDR2` folder at the top level of the repo for archiving purpose. This way, links to specific commits that are tied to the validation steps of specific data release will still be there, and we will also have an easy way to publish did-and-done tests.

**Comment by jsdillon** [comment]
I like the current structure with attaching milestones to represent datasets/full-analyses. It seems like a better way to track progress, IMO, and it avoids duplication of shared bits of validation.
